### PR TITLE
Creating a Middleware

### PR DESCRIPTION
**Pull request by s-laugh**
## [DCWC-1692](https://jira-dev.bdm-dev.dts-stn.com/browse/DCWC-1692) (Jira Issue)

This should only be moving code, not changing it.

Removing the UniversalBenefit.js and putting the chunks of code in better places.

### CODE REVIEW

**Review by DaceyTom2 — CHANGES_REQUESTED** [review]

File: pages/api/programData/_middleware.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { TASK_GROUPS } from '../../../contents/DashboardBenefitTasksConstants'
+
+export function CreateGenericBenefitObj(
```
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/api/programData/_middleware.js]: move to lib - will re-add _middleware.js when auth is merged

File: pages/api/programData/ei.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    status: StatusCodes.inPayment,
+  },
+]
+
+export const EITypes = [
+  {
+    value: 1,
+    type: TypeCodes.EIUnknown,
+  },
+]
```
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/api/programData/ei.js]: Move to it's own consts file.

File: pages/api/programData/ei.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  try {
+    const eiData = await fetch(process.env.EI_ACTIVE_BENEFIT_URL, {
+      headers: new Headers({
+        'Ocp-Apim-Subscription-Key': process.env.OCP_APIM_SUBSCRIPTION_KEY,
+      }),
+    })
+    return CreateBenefitObjWithEIData(await eiData.json())
+  } catch (e) {
+    console.log(e)
+    return null
```
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/api/programData/ei.js:13]: Not sure about that... This isn't actually used as an API call as it's called from 'GetServerSideProps' which is on the same server; doesn't make sense for it to call itself; instead this current implementation is just a function reference.

File: pages/dashboard.js [hunk 4]
```diff
@@ -201,71 +196,32 @@ export default function Dashboard(props) {
...
-
-  const activeCpp = await getActiveCpp()
-  const activeEi = await getActiveEi()
-  currentBenefits.push(CreateBenefitObjWithCPPData(activeCpp))
-  currentBenefits.push(CreateBenefitObjWithEIData(activeEi))
+  const usersBenefts = [] // to be retrieved by API
+  const cppData = await GetCPPProgramData()
+  if (cppData) usersBenefts.push(cppData)
+  const eiData = await GetEIProgramData()
+  if (eiData) usersBenefts.push(eiData)
```
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/dashboard.js:212]: Yes.... yes it should. Would it be alright to manage that in another PR?

[the remark this answers is not in the document]
Comment (DaceyTom2) [reply pages/dashboard.js:212]: Sure thing

File: pages/api/programData/cpp.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import { ProgramCodes } from '../../../constants/ProgramCodes'
+import { TypeCodes } from '../../../constants/ProgramTypeCodes'
+import { StatusCodes } from '../../../constants/StatusCodes'
+import { SummaryTypes } from '../../../constants/SummaryTypes'
+import { CreateBenefitSummary, CreateGenericBenefitObj } from './_middleware'
+
+export default async function GetCPPProgramData() {
```
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/api/programData/cpp.js:3]: solving when moved to proper API in another PR

File: pages/api/programData/ei.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (s-laugh) [reply pages/api/programData/ei.js:13]: Yup, will do in another PR

**Review by DaceyTom2 — APPROVED** [review]
Looks good!

**Pull request merged**